### ISSUE TITLE
[QUERY] Distributed Tracing in EventHub - example?

### ISSUE THREAD

**Issue by sebader** [Azure/azure-sdk-for-net#14201]
**Query/Question**
Now that EventHub supports distributed tracing and submits the Diagnostics-id in a similar way to what Service Bus already did, do you have a corresponding example how to use that ID in the processor? This show it for Service Bus messages: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-end-to-end-tracing#trace-message-processing
````
private readonly TelemetryClient telemetryClient;

async Task ProcessAsync(Message message)
{
    var activity = message.ExtractActivity();
    
    // If you are using Microsoft.ApplicationInsights package version 2.6-beta or higher, you should call StartOperation<RequestTelemetry>(activity) instead
    using (var operation = telemetryClient.StartOperation<RequestTelemetry>("Process", activity.RootId, activity.ParentId))
    {
        telemetryClient.TrackTrace("Received message");
        try 
        {
           // process message
        }
        catch (Exception ex)
        {
             telemetryClient.TrackException(ex);
             operation.Telemetry.Success = false;
             throw;
        }

        telemetryClient.TrackTrace("Done");
   }
}
````

**Environment:**
 - Name and version of the Library package used:   
    - Azure.Messaging.EventHubs.Processor 5.2.0-preview.1
    - Microsoft.ApplicationInsights 2.15.0-beta2
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): [e.g. Azure AppService or Windows 10 .NET Framework 4.8] .NET Core 3.1

**Comment by sebader** [comment]
After some trial-and-error I came up with this. Seems to work when looking at AppInsights but is this really the best way?

````
Activity activity = null;
RequestTelemetry requestTelemetry = null;
if (eventData.Properties.TryGetValue("Diagnostic-Id", out object diagnosticsId))
{
	activity = new Activity("Process Event Hub message");
	activity.SetParentId(diagnosticsId.ToString());
}
else
{
	requestTelemetry = new RequestTelemetry { Name = "Process Event Hub message" };
}

using (var operation = activity != null? _telemetryClient.StartOperation<RequestTelemetry>(activity) : _telemetryClient.StartOperation(requestTelemetry)){
  ...
}
````

**Comment by jsquire** [comment]
Hi @sebader.   To be transparent, I'm not sure what the past path would be to help get you an authoritative answer to this.  We may need to ask a couple of folks for help before finding the right person.  Please bear with us.

Diagnostics for the new generation of Azure SDKs are intended to be a consistent and drive from a common initiative around OpenTelemetry.   Since a good deal of the efforts here were coordinated through Azure.Core, I'm going to start by asking @pakrym for his thoughts or help routing to someone better able to assist.

**Comment by pakrym** [comment]
You are right that we are missing `ExtractActivity` from Track 2 APIs. The workaround you've found looks right but I think we should still have something like ExtractActivity helper.

**Comment by oising** [comment]
hey @pakrym  -- any update here? The docs for service bus look good, but there's nothing equivalent for event hubs (though I imagine they would very, very similar?)

**Comment by pakrym** [comment]
Libraries released starting with this week will have experimental ActivitySource support and OpenTelemetry collection. Initial docs coming in https://github.com/Azure/azure-sdk-for-net/pull/24324
